syntax = "proto3";

package CSR;

message CSTBS{
	int64 deviceID = 1;
	int64 orgID = 2;
	string curveID = 3;
    string hashID = 4;
    int64 pubKLen = 5;
}	

message MCSR{
	int64 deviceID = 1;
	int64 orgID = 2;
	string curveID = 3;
    string hashID = 4;
    int64 pubKLen = 5;
    int64 sigL = 6;
    bytes sig = 7;
}	

message CA{
	int64 deviceID = 1;
	int64 orgID = 2;
	string curveID = 3;
    string hashID = 4;
    int64 pubKLen = 5;
    int64 certSNO = 6;
    int64 caID = 7;
    int64 validF = 8;
    int64 validFor = 9;
}	